standard/statements.md

```diff
@@ -625,93 +623,87 @@ Multiple labels are permitted in a *switch_section*.
 > *end example*
 <!-- markdownlint-disable MD028 -->
 
-<!-- markdownlint-enable MD028 -->
-> *Note*: The “no fall through” rule prevents a common class of bugs that occur in C and C++ when `break` statements are accidentally omitted. For example, the sections of the `switch` statement above can be reversed without affecting the behavior of the statement:
+The *statement_list*s of a *switch_block* may contain declaration statements ([§12.6](statements.md#126-declaration-statements)). The scope of a local variable or constant declared in a switch block is the switch block.
+
+Within a switch block, the meaning of a name used in an expression context shall always be the same ([§11.7.4](expressions.md#1174-simple-names)).
+
+The statement list of a given switch section is reachable if the `switch` statement is reachable and at least one of the following is true:
+
+- The switch expression value matches a `case` label in the switch section.
+- The switch expression value doesn't match any `case` label, and the switch section contains the `default` label.
+- A switch label of the switch section is referenced by a reachable `goto case` or `goto default` statement.
+
+The end point of a `switch` statement is reachable if at least one of the following is true:
+
+- The `switch` statement contains a reachable `break` statement that exits the `switch` statement.
+- The `switch` statement is reachable, the switch expression is a non-constant value, and no `default` label is present.
```

Comment:
This list isn't correct for several reasons:
- It only describes the behavior of constant labels.  What about `case int x:`?  `case var _:`?
- It isn't even correct for constant values.  If you switch on a value of type `byte`, and have cases to handle all values, a `default:` label does not lead to the reachability of its switch section.